marklogic-data-hub/src/test/java/com/marklogic/hub/hubcentral/migration/AllStepsReferencedTest.java

```diff
@@ -0,0 +1,429 @@
+        JsonNode customStep2 = readJsonObject(customSteps.resolve("generate-dictionary.step.json").toFile());
+        JsonNode customStep3 = readJsonObject(customSteps.resolve("custom-mapping-step.step.json").toFile());
+
+        assertEquals(List.of("master-customer", "Customer", "custom-mastering" ), getCollectionsAsList(customStep1));
+        assertEquals(List.of("generate-dictionary", "Customer"), getCollectionsAsList(customStep2));
+        assertEquals(List.of("custom-mapping-step" ), getCollectionsAsList(customStep3));
+
+        verifyOptions(customStep1, mapper.valueToTree(customMasterFlow.getStep("3").getOptions()));
+        verifyOptions(customStep2, mapper.valueToTree(customMasterFlow.getStep("1").getOptions()));
+        verifyOptions(customStep3, mapper.valueToTree(customOnlyFlow.getStep("1").getOptions()));
+
+        assertEquals("custom-mastering", customStep1.get("name").asText());
+        assertEquals("This is a custom mastering step", customStep1.get("description").asText());
+        assertEquals("custom", customStep1.get("stepDefinitionType").asText().toLowerCase());
+        assertEquals("Customer", customStep1.get("targetEntityType").asText());
+        assertEquals("json", customStep1.get("targetFormat").asText());
+        assertEquals("custom-mastering-custom", customStep1.get("stepId").asText());
+
+        assertEquals("generate-dictionary", customStep2.get("name").asText());
+        assertEquals("Generate dictionary custom step", customStep2.get("description").asText());
+        assertEquals("custom", customStep2.get("stepDefinitionType").asText().toLowerCase());
+        assertEquals("Customer", customStep2.get("targetEntityType").asText());
+        assertEquals("json", customStep2.get("targetFormat").asText());
+        assertEquals("generate-dictionary-custom", customStep2.get("stepId").asText());
+
+        assertEquals("custom-mapping-step", customStep3.get("name").asText());
+        assertEquals("maps and harmonizes XML docs to data-hub-FINAL", customStep3.get("description").asText());
+        assertEquals("custom", customStep3.get("stepDefinitionType").asText().toLowerCase());
+        assertEquals("xml", customStep3.get("targetFormat").asText());
+        //"mapping" is copied over in case of custom mapping step with no valid mapping
+        assertNotNull(customStep3.get("mapping"));
+        assertEquals("custom-mapping-step-custom", customStep3.get("stepId").asText());
+    }
+
+    private List getCollectionsAsList(JsonNode customStep){
+        JsonNode collectionsNode = customStep.get("collections");
+        List<String> collectionList = new ArrayList<>();
+        for (JsonNode node : collectionsNode) {
+            collectionList.add(node.asText());
+        }
+        return collectionList;
+    }
+
+    private void verifyMappingSteps(HubProject hubProject, Map<String, Mapping> mappingMap, Map<String, Flow> flowMap) {
```

Comment:
I think it would be good to have similar methods for mastering, matching and merging steps and make sure that the files are created and looks as expected.